lib/topics_filter.rb

```diff
@@ -84,6 +86,15 @@ def filter_status(status:, category_id: nil)
 
   private
 
+  def filter_state(state:)
+    case state
+    when "pinned"
+      @scope.where("topics.pinned_at IS NOT NULL AND topics.pinned_until > topics.pinned_at")
```

Comment:
Just to be sure, we are looking for topics which are pinned or were pinned in the past. Not only topics which are currently pinned, right?